src/components/Calendar/hooks/useRangeStartDate.js

```diff
@@ -0,0 +1,9 @@
+import { useMemo } from 'react';
+import { isSameDay, isEmptyRange } from '../helpers';
+
+export default function useRangeStartDate(date, currentRange) {
+    return useMemo(() => (!isEmptyRange(currentRange) ? isSameDay(date, currentRange[0]) : false), [
```

Comment:
to improve readability we need to avoid negate conditions instead we could:
```isEmptyRange(currentRange) ? false : isSameDay(date, currentRange[0])```